src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindCollectionsOperation.java

```diff
@@ -0,0 +1,54 @@
+package io.stargate.sgv2.jsonapi.service.operation.model.impl;
+
+import io.smallrye.mutiny.Uni;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandContext;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.service.bridge.executor.QueryExecutor;
+import io.stargate.sgv2.jsonapi.service.operation.model.Operation;
+import io.stargate.sgv2.jsonapi.service.schema.CollectionManager;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Find collection operation. Uses {@link CollectionManager} to fetch all valid jsonapi tables for a
+ * namespace.
+ *
+ * @param collectionManager {@link CollectionManager}
+ * @param commandContext {@link CommandContext}
+ */
+public record FindCollectionsOperation(
+    CollectionManager collectionManager, CommandContext commandContext) implements Operation {
```

Comment:
add test for this class

Reply:
No need for a test, all cases tested in the integration test. The test would anyway be very hard to create, as I would need a full table schema for a matching table..